docs/src/components/SandpackContainerReact.astro

```diff
@@ -0,0 +1,168 @@
+---
+/** @jsxImportSource react */
+import Sandpack from './SandpackBase';
+
+let {
+  editorHeight = 230,
+  previewAspectRatio = 16 / 9,
+  stacked,
+  reversed,
+  ...props
+} = Astro.props as Props;
+
+if (typeof editorHeight === 'number') editorHeight = `${editorHeight}px`;
+---
+<!--
+  @TODO This is basically a duplicate of SandpackContainer.astro but uses SandpackBase instead of the more opinionated Sandpack React component.
```

Comment:
NOTE: This todo is a callout on the refactor to support sandpack templates like `'vite-react-ts'` for react cases without a full rearchitecture to also handle convenience crud for standard web component use cases/backwards compatibility. Didn't want to blow up effort here, but if someone else sees a quick+clean alternative solution, happy to accept contribs.